java/ql/lib/semmle/code/java/frameworks/Properties.qll

```diff
@@ -1,15 +1,25 @@
 /* Definitions related to `java.util.Properties`. */
 import semmle.code.java.Type
+private import semmle.code.java.dataflow.FlowSteps
 
 library class TypeProperty extends Class {
   TypeProperty() { hasQualifiedName("java.util", "Properties") }
 }
 
-library class PropertiesGetPropertyMethod extends Method {
+library class PropertiesGetPropertyMethod extends ValuePreservingMethod {
   PropertiesGetPropertyMethod() {
     getDeclaringType() instanceof TypeProperty and
     hasName("getProperty")
   }
+
+  override predicate returnsValue(int arg) { arg = 1 }
+}
+
+library class PropertiesGetMethod extends Method {
```

Comment:
Don't use `library` in new code

Reply:
I did it for consistency with the rest of the file, should I drop `library` from the other classes in this file? I'd like them to be consistent.

Reply:
Just don't add new code using it; we'll get around to a bulk cleanup at some point.